src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/service/events/InboundEvents.kt

```diff
@@ -0,0 +1,68 @@
+package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.service.events
+
+import com.fasterxml.jackson.annotation.JsonGetter
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import java.time.Instant
+
+enum class InboundEventType(val eventType: String) {
+  CIAG_INDUCTION_CREATED("ciag-induction.created") {
+    override fun toInboundEvent(mapper: ObjectMapper, message: String) =
+      mapper.readValue<CiagInductionCreatedEvent>(message)
+  },
+  CIAG_INDUCTION_UPDATED("ciag-induction.updated") {
+    override fun toInboundEvent(mapper: ObjectMapper, message: String) =
+      mapper.readValue<CiagInductionUpdatedEvent>(message)
+  },
+  ;
+
+  abstract fun toInboundEvent(mapper: ObjectMapper, message: String): InboundEvent
+}
+
+interface InboundEvent {
+  fun eventType(): String
+  fun reference(): String
+  fun prisonNumber(): String
+  fun occurredAt(): Instant
+}
+
+// ------------ CIAG Induction events ------------------------------------------
+abstract class CiagInductionEvent(
```

Comment:
I've copied this approach from activities-management-api. It might seem a bit complex at first glance, but this structure gives us flexibility to extend this for other `InboundEvents` of interest in the future.

Reply:
I know you've copied it, and its a pattern in use by other projects, so there's a lot to be said for that ... but it seems really complicated 😢

Reply:
Why is it an abstract class rather than a concrete class?